pkg/combination/combination.go

```diff
@@ -37,6 +49,11 @@ func NewStream(options ...StreamOption) Stream {
 	for _, option := range options {
 		option(cs)
 	}
+	for key := range cs.args {
+		cs.parameterListFromArgs = append(cs.parameterListFromArgs, key)
+		cs.positionsMapInArgs = append(cs.positionsMapInArgs, 0)
+	}
+	cs.lastUpdatedParameter = len(cs.parameterListFromArgs) - 2
```

Comment:
What's the significance of the `2` here? Is this a potential [magic number](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad#:~:text=In%20programming%2C%20a%20%22magic%20number,in%20more%20than%20one%20place.)?

Reply:
The `len(cs.parameterListFromArgs) - 2` means we are setting lastUpdatedParameter to the second to last variable in the list of parameterListFromArgs. We do not want to look at the last variable in the list, in my logic for` i = len(cs.positionsMapInArgs) - 1; i > cs.lastUpdatedParameter; i--` you can it goes too `>`. If it would be more intiutive, I think we could change it to `len(cs.parameterListFromArgs) - 1` and make changes elsewhere. It just made more sense when I was writing it to start it at the second to last value in the list.

Reply:
I haven't dived too much into the implementation, but I think my preference (generally speaking) is to use a constant variable or adding a comment as to why we're performing that operation. That way, the code is a bit easier to read without having to take a second trying to reason about why a number if being used.

Reply:
@timflannagan I will go ahead and add a comment. I also think I will go ahead a rename the variable to `nextParameterToUpdate` as this will be more intuitive on why we are looking at the second to last item and not the actual last item.